packages/evolution-backend/src/models/interviews.db.queries.ts

```diff
@@ -26,17 +30,51 @@ export interface InterviewSearchAttributes {
     username?: string | undefined;
     facebook: boolean;
     google: boolean;
+    surveyId?: number;
 }
 
+/** this will return the survey id or create a new survey in
+ * table sv_surveys if no existing survey shortname matches
+ * the project shortname
+ */
+const getSurveyId = async (): Promise<number> => {
+    if (_surveyId) {
+        return _surveyId;
+    }
+
+    // check in survey table if at least one matches the project shortname:
+    const matchingSurvey = await knex.select('*').from(surveyTable).where('shortname', config.projectShortname);
+
+    if (matchingSurvey.length === 1) {
+        return matchingSurvey[0].id;
+    }
+
+    // no match: create a new survey
+    const returning = await knex(surveyTable)
```

Comment:
Here you can use the onConflict knex function, with ignore. The _surveyId will probably be set from another thread